bench/utils/__init__.py

```diff
@@ -420,10 +421,14 @@ def fetch_details_from_tag(_tag: str) -> Tuple[str, str, str]:
 	return org, repo, tag
 
 
-def is_git_url(url: str) -> bool:
-	# modified to allow without the tailing .git from https://github.com/jonschlinkert/is-git-url.git
-	pattern = r"(?:git|ssh|https?|\w*@[-\w.]+):(\/\/)?(.*?)(\.git)?(\/?|\#[-\d\w._]+?)$"
-	return bool(re.match(pattern, url))
+def parse_git_url(url: str):
+	try:
+		parsed_obj = giturlparse.parse(url)
+	except Exception:
+		# the git url is not parsable
+		return False
```

Comment:
Why not let it just raise instead? That's what I'd expect from a util named `parse_git_url`.

Reply:
if we look at how it's being [used](https://github.com/frappe/bench/pull/1233/files#diff-3722f4ad716221f5c6d162bf82efd37d96c6367678903aba4e51725dd57ee75bR80) - we'll see that we pass everything coming our way into `parse_git_url` ..so if we raise everytime we'll need to handle that. should we?

Reply:
The problem I was trying to address here was a naming one. Everything could stay wherever they are if this function is renamed to `get_parsed_giturl` and returns `None` instead of `False`.

Reply:
ah sorry i misinterpreted 😅 - will do 👍